client/src/pages/NewReport/ApplicationSection/ApplicationSection.tsx

```diff
@@ -0,0 +1,127 @@
+        added: '07.03.2024 15:32',
+    },
+];
+
+const ApplicationSection = () => {
+    const [rows, setRows] = useState<ApplicationEntry[]>(MOCK_APPLICATIONS);
+    const [showModal, setShowModal] = useState(false);
+
+    const columns: Array<TableColumn<ApplicationEntry>> = [
+        {
+            header: 'Name',
+            columnKey: 'name',
+            customComponent: (row: ApplicationEntry) => (
+                <a href="#" className="application-section__link">
+                    {row.name}
+                </a>
+            ),
+        },
+        {
+            header: 'Precision',
+            columnKey: 'precision',
+            customComponent: (row: ApplicationEntry) => (
+                <TagButton
+                    listItems={['high', 'medium', 'low', 'none']}
+                    chosenItem={row.precision}
+                    onSelect={(precision) => handlePrecisionChange(row.id, precision)}
+                />
+            ),
+        },
+        {
+            header: 'Custom prompt',
+            columnKey: 'customPrompt',
+            customComponent: (row: ApplicationEntry) => (
+                <input
+                    type="text"
+                    className="application-section__input"
+                    value={row.customPrompt}
+                    onChange={(e) => handleCustomPromptChange(row.id, e.target.value)}
+                />
+            ),
+        },
+        { header: 'Updated', columnKey: 'updated' },
+        { header: 'Added', columnKey: 'added' },
+        {
+            header: 'Actions',
+            columnKey: 'actions',
+            customComponent: (row: ApplicationEntry) => (
+                <ActionButton
+                    onClick={() => handleDelete(row.id)}
+                    description="Delete"
+                    color={ActionButtonColor.RED}
+                />
+            ),
+        }
+    ];
+
+    const handlePrecisionChange = (id: string, precision: string) => {
+        setRows((prevRows) =>
+            prevRows.map((row) =>
+                row.id === id ? { ...row, precision } : row
+            )
+        );
+    };
+
+    const handleCustomPromptChange = (id: string, customPrompt: string) => {
+        setRows((prevRows) =>
+            prevRows.map((row) =>
+                row.id === id ? { ...row, customPrompt } : row
+            )
+        );
+    };
+
+    const handleDelete = (id: string) => {
+        setRows((prevRows) => prevRows.filter((row) => row.id !== id));
+    };
+
+    const handleAddClick = () => {
+        setShowModal(true);
+    };
+
+    const handleCloseModal = () => {
+        setShowModal(false);
+    };
+
+    return (
+        <SectionComponent
+            icon={<SVGIcon iconName='application-icon' />}
+            title={'Applications'}
+            callback={handleAddClick}>
+            {showModal && <OverlayComponent onClose={handleCloseModal} />}
+            <div className="application-section__table">
```

Comment:
Is this class (application-section__table) necessary? How is this table different from the other ones?
Maybe if there is something wrong with the styles we should consider chaning SectionComponent or Table?